### PR TITLE
Consolidate utils/veda-data(-no-faux-module) methods into new dir and remove faux module dependencies

### PR DESCRIPTION
**Pull request by sandrahoang686**
**Follow-up clean up PR to https://github.com/NASA-IMPACT/veda-ui/pull/976**

**Preview for GHG instance with this branch** 
https://github.com/US-GHG-Center/veda-config-ghg/pull/430
https://deploy-preview-430--ghg-demo.netlify.app/

**Related Ticket:** https://github.com/NASA-IMPACT/veda-ui/issues/800
### Description of Changes
* Removes old `utils/veda-data.ts` and `utils/veda-data-no-faux-module.ts` files to be consolidated under new dir `utils/veda-data/`
* Removes all veda faux modules dependencies from `utils/veda-data/` by making methods take in datasets/stories as params instead

### Notes & Questions About Changes
_{Add additonal notes and outstanding questions here related to changes in this pull request}_

### Validation / Testing
- [ ] Make sure everything in app still works as is
- [ ] Make sure ghg and other instances still work as is with this branch
- [ ] Make sure build still works for nextJs instance

### CODE REVIEW

File: app/scripts/utils/veda-data/index.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+export const getAllDatasetsProps = (datasets: VedaData<DatasetData>) => Object.values(datasets).map((d) => d!.data);
+
+/**
+ * List with the meta information of all stories.
+ */
+// @QUESTION: This was migrated over from the original /scripts/utils/veda-data file but doesn't seem to be used anywhere, we should get rid of it I think
+// Also it does the same thing as getAllDatasetsProps
+export const getAllStoriesProps = (stories: VedaData<StoryData>) => Object.values(stories).map(
+  (d) => d!.data
+);
```
Comment (sandrahoang686) [inline app/scripts/utils/veda-data/index.ts:20]: We can probably get rid of this as it wasn't used in the original file either

File: app/scripts/components/exploration/data-utils.ts [hunk 2]
```diff
@@ -39,14 +39,10 @@ export const findDatasetAttribute = ({
   datasetId: string;
   attr: string;
 }) => {
-  return veda_datasets[datasetId]?.data[attr];
+  return veda_faux_module_datasets[datasetId]?.data[attr];
 };
 
-export const allDatasets = Object.values(veda_datasets)
```
Comment (sandrahoang686) [inline app/scripts/components/exploration/data-utils.ts:45]: Same logic as `getAllDatasetsProps` so removing

File: app/scripts/utils/veda-data/index.ts (hunk 1 above)
Comment (dzole0311) [reply app/scripts/utils/veda-data/index.ts:20]: +1 for removing if not used

**Comment by sandrahoang686** [comment]
**Preview for GHG instance with this branch** 
https://github.com/US-GHG-Center/veda-config-ghg/pull/430
https://deploy-preview-430--ghg-demo.netlify.app/

File: app/scripts/components/analysis/results/chart-card.tsx [hunk 3]
```diff
@@ -117,8 +118,9 @@ const getNoDownloadReason = ({ status, data }: TimeseriesData) => {
  * @returns Internal path for Link
  */
 const getDatasetOverviewPath = (layerId: string) => {
+  const allDatasetsProps = getAllDatasetsProps(veda_faux_module_datasets);
   const dataset = allDatasetsProps.find((d) =>
-    d.layers.find((l) => l.id === layerId)
+    d?.layers.find((l) => l.id === layerId)
```
Comment (hanbyul-here) [inline app/scripts/components/analysis/results/chart-card.tsx:123]: any reason that optional chain is needed here?

Comment (sandrahoang686) [reply app/scripts/components/analysis/results/chart-card.tsx:123]: I dont remember why i added that optional chain... but its safer with 😄

**Pull request merged**